File: proyecto_pedidos/suscriptor2/datos/ConexionRabbit.java
package datos;

import com.rabbitmq.client.*;
import negocio.Pedido;
import presentacion.VentanaSuscriptor;
import org.json.JSONObject;

public class ConexionRabbit {
    private static final String EXCHANGE = "pedidos_exchange";
    private static final String QUEUE = "cola_gui";
    private static final String ROUTING_KEY = "pedidos_key";
    private static final String HOST = "shark.rmq.cloudamqp.com";
    private static final String USER = "ojbcnehb";
    private static final String PASSWORD = "xJw2qw5O8E7sVYTEYqP3u0aW_HfkjVvo";
    private static final String VHOST = "ojbcnehb";

    public void conectarYEscuchar(VentanaSuscriptor ventana) throws Exception {
        ConnectionFactory factory = new ConnectionFactory();
        factory.setHost(HOST);
        factory.setUsername(USER);
        factory.setPassword(PASSWORD);
        factory.setVirtualHost(VHOST);
        factory.setPort(5672);

        Connection connection = factory.newConnection();
        Channel channel = connection.createChannel();

        channel.exchangeDeclare(EXCHANGE, "direct", true);
        channel.queueDeclare(QUEUE, true, false, false, null);
        channel.queueBind(QUEUE, EXCHANGE, ROUTING_KEY);

        channel.basicQos(1);

        DeliverCallback deliverCallback = (consumerTag, delivery) -> {
            String mensaje = new String(delivery.getBody(), "UTF-8");
            System.out.println("📨 Mensaje recibido: " + mensaje);

            JSONObject json = new JSONObject(mensaje);
            Pedido pedido = new Pedido(
                json.getString("sku"),
                json.getInt("cantidad"),
                json.getString("talla"),
                json.getString("color")
            );

            ventana.mostrarPedido(pedido);
            channel.basicAck(delivery.getEnvelope().getDeliveryTag(), false);
        };

        channel.basicConsume(QUEUE, false, deliverCallback, consumerTag -> {});
        System.out.println("Suscriptor 2 escuchando mensajes");
    }
}

File: proyecto_pedidos/suscriptor1/datos/conexion_rabbit.py
import pika

class ConexionRabbit:
    EXCHANGE = "pedidos_exchange"
    QUEUE = "cola_sql"
    ROUTING_KEY = "pedidos_key"
    
    def __init__(self):
        self.connection = None
        self.channel = None
    
    def conectar(self):
        credentials = pika.PlainCredentials(
            username="ojbcnehb",
            password="xJw2qw5O8E7sVYTEYqP3u0aW_HfkjVvo"
        )
        parameters = pika.ConnectionParameters(
            host="shark.rmq.cloudamqp.com",
            port=5672,
            virtual_host="ojbcnehb",
            credentials=credentials
        )
        self.connection = pika.BlockingConnection(parameters)
        self.channel = self.connection.channel()
        
        self.channel.exchange_declare(
            exchange=self.EXCHANGE,
            exchange_type="direct",
            durable=True
        )
        
        self.channel.queue_declare(queue=self.QUEUE, durable=True)
        
        self.channel.queue_bind(
            exchange=self.EXCHANGE,
            queue=self.QUEUE,
            routing_key=self.ROUTING_KEY
        )
        return True
    
    def escuchar(self, callback):
        self.channel.basic_qos(prefetch_count=1)
        self.channel.basic_consume(
            queue=self.QUEUE,
            on_message_callback=callback
        )
        print("Suscriptor 1 escuchando los mensajes")
        self.channel.start_consuming()
    